src/main/java/com/fasterxml/jackson/databind/ser/std/NumberSerializer.java

```diff
@@ -69,6 +81,10 @@ public JsonSerializer<?> createContextual(SerializerProvider prov,
     @Override
     public void serialize(Number value, JsonGenerator g, SerializerProvider provider) throws IOException
     {
+        if (format != null) {
+            g.writeString(format.format(value));
```

Comment:
Should probably use `g.writeNumber(String)` instead, to indicate "it is a number but pre-formatted"; avoids

Reply:
@cowtowncoder It looks like `g.writeNumber(String)` writes without quotes. But formatted numbers have to be quoted. Don't they?

Reply:
Correct, that is the difference: whether output is JSON Number (no quotes) or JSON String (quotes).

It really depends on what the goal is -- if it is to, say, limit precision to certain number of fractional digits, probably not. But if it is to format with different thousand separators, sure.

And this is why I think `shape` has to be considered: if it's `Shape.STRING` then yes, output should be as String.